// ng2-clipboard/bundles/ng2-clipboard.umd.js
!function(e){function r(e,r,o){return 4===arguments.length?t.apply(this,arguments):void n(e,{declarative:!0,deps:r,declare:o})}function t(e,r,t,o){n(e,{declarative:!1,deps:r,executingRequire:t,execute:o})}function n(e,r){r.name=e,e in v||(v[e]=r),r.normalizedDeps=r.deps}function o(e,r){if(r[e.groupIndex]=r[e.groupIndex]||[],-1==g.call(r[e.groupIndex],e)){r[e.groupIndex].push(e);for(var t=0,n=e.normalizedDeps.length;n>t;t++){var a=e.normalizedDeps[t],u=v[a];if(u&&!u.evaluated){var d=e.groupIndex+(u.declarative!=e.declarative);if(void 0===u.groupIndex||u.groupIndex<d){if(void 0!==u.groupIndex&&(r[u.groupIndex].splice(g.call(r[u.groupIndex],u),1),0==r[u.groupIndex].length))throw new TypeError("Mixed dependency cycle detected");u.groupIndex=d}o(u,r)}}}}function a(e){var r=v[e];r.groupIndex=0;var t=[];o(r,t);for(var n=!!r.declarative==t.length%2,a=t.length-1;a>=0;a--){for(var u=t[a],i=0;i<u.length;i++){var s=u[i];n?d(s):l(s)}n=!n}}function u(e){return y[e]||(y[e]={name:e,dependencies:[],exports:{},importers:[]})}function d(r){if(!r.module){var t=r.module=u(r.name),n=r.module.exports,o=r.declare.call(e,function(e,r){if(t.locked=!0,"object"==typeof e)for(var o in e)n[o]=e[o];else n[e]=r;for(var a=0,u=t.importers.length;u>a;a++){var d=t.importers[a];if(!d.locked)for(var i=0;i<d.dependencies.length;++i)d.dependencies[i]===t&&d.setters[i](n)}return t.locked=!1,r},{id:r.name});t.setters=o.setters,t.execute=o.execute;for(var a=0,i=r.normalizedDeps.length;i>a;a++){var l,s=r.normalizedDeps[a],c=v[s],f=y[s];f?l=f.exports:c&&!c.declarative?l=c.esModule:c?(d(c),f=c.module,l=f.exports):l=p(s),f&&f.importers?(f.importers.push(t),t.dependencies.push(f)):t.dependencies.push(null),t.setters[a]&&t.setters[a](l)}}}function i(e){var r,t=v[e];if(t)t.declarative?f(e,[]):t.evaluated||l(t),r=t.module.exports;else if(r=p(e),!r)throw new Error("Unable to load dependency "+e+".");return(!t||t.declarative)&&r&&r.__useDefault?r["default"]:r}function l(r){if(!r.module){var t={},n=r.module={exports:t,id:r.name};if(!r.executingRequire)for(var o=0,a=r.normalizedDeps.length;a>o;o++){var u=r.normalizedDeps[o],d=v[u];d&&l(d)}r.evaluated=!0;var c=r.execute.call(e,function(e){for(var t=0,n=r.deps.length;n>t;t++)if(r.deps[t]==e)return i(r.normalizedDeps[t]);throw new TypeError("Module "+e+" not declared as a dependency.")},t,n);void 0!==c&&(n.exports=c),t=n.exports,t&&t.__esModule?r.esModule=t:r.esModule=s(t)}}function s(r){var t={};if(("object"==typeof r||"function"==typeof r)&&r!==e)if(m)for(var n in r)"default"!==n&&c(t,r,n);else{var o=r&&r.hasOwnProperty;for(var n in r)"default"===n||o&&!r.hasOwnProperty(n)||(t[n]=r[n])}return t["default"]=r,x(t,"__useDefault",{value:!0}),t}function c(e,r,t){try{var n;(n=Object.getOwnPropertyDescriptor(r,t))&&x(e,t,n)}catch(o){return e[t]=r[t],!1}}function f(r,t){var n=v[r];if(n&&!n.evaluated&&n.declarative){t.push(r);for(var o=0,a=n.normalizedDeps.length;a>o;o++){var u=n.normalizedDeps[o];-1==g.call(t,u)&&(v[u]?f(u,t):p(u))}n.evaluated||(n.evaluated=!0,n.module.execute.call(e))}}function p(e){if(I[e])return I[e];if("@node/"==e.substr(0,6))return I[e]=s(D(e.substr(6)));var r=v[e];if(!r)throw"Module "+e+" not present.";return a(e),f(e,[]),v[e]=void 0,r.declarative&&x(r.module.exports,"__esModule",{value:!0}),I[e]=r.declarative?r.module.exports:r.esModule}var v={},g=Array.prototype.indexOf||function(e){for(var r=0,t=this.length;t>r;r++)if(this[r]===e)return r;return-1},m=!0;try{Object.getOwnPropertyDescriptor({a:0},"a")}catch(h){m=!1}var x;!function(){try{Object.defineProperty({},"a",{})&&(x=Object.defineProperty)}catch(e){x=function(e,r,t){try{e[r]=t.value||t.get.call(e)}catch(n){}}}}();var y={},D="undefined"!=typeof System&&System._nodeRequire||"undefined"!=typeof require&&"undefined"!=typeof require.resolve&&"undefined"!=typeof process&&process.platform&&require,I={"@empty":{}};return function(e,n,o,a){return function(u){u(function(u){for(var d={_nodeRequire:D,register:r,registerDynamic:t,get:p,set:function(e,r){I[e]=r},newModule:function(e){return e}},i=0;i<n.length;i++)(function(e,r){r&&r.__esModule?I[e]=r:I[e]=s(r)})(n[i],arguments[i]);a(d);var l=p(e[0]);if(e.length>1)for(var i=1;i<e.length;i++)p(e[i]);return o?l["default"]:l})}}}("undefined"!=typeof self?self:global)

(["1"], ["4","3"], true, function($__System) {
var require = this.require, exports = this.exports, module = this.module;
$__System.registerDynamic("2", ["3", "4", "5"], true, function ($__require, exports, module) {
    "use strict";

    var global = this || self,
        GLOBAL = global;
    Object.defineProperty(exports, "__esModule", { value: true });
    var core_1 = $__require("3");
    var common_1 = $__require("4");
    var font_awesome_component_1 = $__require("5");
    var FontAwesomeModule = function () {
        function FontAwesomeModule() {}
        return FontAwesomeModule;
    }();
    FontAwesomeModule.decorators = [{ type: core_1.NgModule, args: [{
            imports: [common_1.CommonModule],
            declarations: [font_awesome_component_1.FontAwesomeComponent],
            exports: [font_awesome_component_1.FontAwesomeComponent]
        }] }];
    /** @nocollapse */
    FontAwesomeModule.ctorParameters = function () {
        return [];
    };
    exports.FontAwesomeModule = FontAwesomeModule;
    
});
$__System.registerDynamic("5", ["3"], true, function ($__require, exports, module) {
    "use strict";

    var global = this || self,
        GLOBAL = global;
    var __decorate = this && this.__decorate || function (decorators, target, key, desc) {
        var c = arguments.length,
            r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc,
            d;
        if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
        return c > 3 && r && Object.defineProperty(target, key, r), r;
    };
    var __metadata = this && this.__metadata || function (k, v) {
        if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
    };
    Object.defineProperty(exports, "__esModule", { value: true });
    var core_1 = $__require("3");
    var FontAwesomeComponent = function () {
        function FontAwesomeComponent() {}
        Object.defineProperty(FontAwesomeComponent.prototype, "faIcon", {
            get: function () {
                return this._faIcon;
            },
            set: function (icon) {
                // sanitize the icon each time it is set.
                // if they didn't provide something like this: 'fa fa-cubes', fix it for them.
                if (icon) {
                    if (!icon.match(/^\s?fa fa\-/i)) {
                        if (!icon.match(/^fa\-/i)) {
                            icon = 'fa-' + icon;
                        }
                        if (!icon.match(/^fa /i)) {
                            icon = 'fa ' + icon;
                        }
                    }
                }
                this._faIcon = icon;
            },
            enumerable: true,
            configurable: true
        });
        // ngOnInit sanitizes faIcon on init
        FontAwesomeComponent.prototype.ngOnInit = function () {};
        return FontAwesomeComponent;
    }();
    __decorate([core_1.Input(), __metadata("design:type", String), __metadata("design:paramtypes", [String])], FontAwesomeComponent.prototype, "faIcon", null);
    FontAwesomeComponent = __decorate([core_1.Component({
        // moduleId: module.id,
        selector: 'font-awesome',
        template: "\n    <span *ngIf=\"faIcon\" [class]=\"faIcon\" aria-hidden=\"true\"></span>\n  ",
        styles: ["\n\n  "]
    })], FontAwesomeComponent);
    exports.FontAwesomeComponent = FontAwesomeComponent;

    
});
$__System.registerDynamic("6", ["2", "5"], true, function ($__require, exports, module) {
  "use strict";

  var global = this || self,
      GLOBAL = global;
  Object.defineProperty(exports, "__esModule", { value: true });
  var font_awesome_module_1 = $__require("2");
  exports.FontAwesomeModule = font_awesome_module_1.FontAwesomeModule;
  var font_awesome_component_1 = $__require("5");
  exports.FontAwesomeComponent = font_awesome_component_1.FontAwesomeComponent;
  
});
$__System.registerDynamic("7", ["3", "4", "8", "9", "6"], true, function ($__require, exports, module) {
    "use strict";

    var global = this || self,
        GLOBAL = global;
    Object.defineProperty(exports, "__esModule", { value: true });
    var core_1 = $__require("3");
    var common_1 = $__require("4");
    var clipboard_service_1 = $__require("8");
    var clipboard_component_1 = $__require("9");
    var ng2_font_awesome_1 = $__require("6");
    var ClipboardModule = function () {
        function ClipboardModule() {}
        return ClipboardModule;
    }();
    ClipboardModule.decorators = [{ type: core_1.NgModule, args: [{
            imports: [common_1.CommonModule, ng2_font_awesome_1.FontAwesomeModule],
            exports: [clipboard_component_1.ClipboardComponent],
            declarations: [clipboard_component_1.ClipboardComponent],
            providers: [clipboard_service_1.ClipboardService]
        }] }];
    /** @nocollapse */
    ClipboardModule.ctorParameters = function () {
        return [];
    };
    exports.ClipboardModule = ClipboardModule;
    
});
$__System.registerDynamic('a', [], true, function ($__require, exports, module) {
    // greg hedin
    "use strict";

    var global = this || self,
        GLOBAL = global;
    var Clipboard = function () {
        function Clipboard() {}
        Clipboard.init = function () {
            window.addEventListener('copy', function (e) {
                if (Clipboard._data) {
                    e.clipboardData.setData('text/plain', Clipboard._data);
                    e.preventDefault(); // this disables 'ctrl + c'
                    Clipboard._data = null;
                }
            });
        };
        Clipboard._data = null;
        // MSIE boolean
        Clipboard.POS = !!window['clipboardData'];
        Clipboard.copy = function (data) {
            Clipboard._data = data;
            if (!Clipboard.POS) {
                document.execCommand('copy');
            } else if (!!window['clipboardData']) {
                window['clipboardData'].setData('Text', Clipboard._data);
            }
        };
        return Clipboard;
    }();
    exports.Clipboard = Clipboard;
    Clipboard.init();
    
});
$__System.registerDynamic("8", ["3", "a"], true, function ($__require, exports, module) {
    "use strict";

    var global = this || self,
        GLOBAL = global;
    Object.defineProperty(exports, "__esModule", { value: true });
    var core_1 = $__require("3");
    var ts_clipboard_1 = $__require("a");
    var ClipboardService = function () {
        function ClipboardService() {}
        // clipboard = {copy: function(str:string) { console.log('copy %s', str); } };
        ClipboardService.prototype.copy = function (str) {
            ts_clipboard_1.Clipboard.copy(str);
        };
        ;
        return ClipboardService;
    }();
    ClipboardService.decorators = [{ type: core_1.Injectable }];
    /** @nocollapse */
    ClipboardService.ctorParameters = function () {
        return [];
    };
    exports.ClipboardService = ClipboardService;
    
});
$__System.registerDynamic("9", ["3", "8"], true, function ($__require, exports, module) {
    "use strict";

    var global = this || self,
        GLOBAL = global;
    Object.defineProperty(exports, "__esModule", { value: true });
    var core_1 = $__require("3");
    var clipboard_service_1 = $__require("8");
    var ClipboardComponent = function () {
        function ClipboardComponent(clipboard) {
            var _this = this;
            this.clipboard = clipboard;
            this.content = ''; // the text to be copied
            this.noClass = false;
            this.copyText = function () {
                if (_this.content) {
                    // console.log('copyText: %s' + this.content);
                    _this.clipboard.copy(_this.content);
                }
            };
        }
        // load default config
        ClipboardComponent.prototype.ngOnInit = function () {
            if (!this.faIcon && !this.img && !this.text) {
                this.text = 'copy';
            }
            if (!this.altText) {
                this.altText = 'copy to clipboard';
            }
        };
        return ClipboardComponent;
    }();
    ClipboardComponent.decorators = [{ type: core_1.Component, args: [{
            // moduleId: module.id,
            selector: 'clipboard-component',
            template: "\n    <span [ngClass]=\"{'clipboard': !noClass}\" (click)=\"copyText()\" [title]=\"altText\">\n      <font-awesome *ngIf=\"faIcon\" [faIcon]=\"'copy'\"></font-awesome>\n      <img *ngIf=\"img\" [src]=\"img\"/>\n    {{text}}</span>\n  ",
            styles: ["\n    .clipboard { cursor: pointer; border-radius: 3px; display: inline-block; padding: 2px; margin: 2px; background-color: #fff; color: #000; box-shadow: 1px 1px 10px black; }\n  "]
        }] }];
    /** @nocollapse */
    ClipboardComponent.ctorParameters = function () {
        return [{ type: clipboard_service_1.ClipboardService }];
    };
    ClipboardComponent.propDecorators = {
        'content': [{ type: core_1.Input }],
        'faIcon': [{ type: core_1.Input }],
        'img': [{ type: core_1.Input }],
        'altText': [{ type: core_1.Input }],
        'text': [{ type: core_1.Input }],
        'noClass': [{ type: core_1.Input }]
    };
    exports.ClipboardComponent = ClipboardComponent;
    
});
$__System.registerDynamic("1", ["7", "8", "9"], true, function ($__require, exports, module) {
  "use strict";

  var global = this || self,
      GLOBAL = global;
  Object.defineProperty(exports, "__esModule", { value: true });
  var clipboard_module_1 = $__require("7");
  exports.ClipboardModule = clipboard_module_1.ClipboardModule;
  var clipboard_service_1 = $__require("8");
  exports.ClipboardService = clipboard_service_1.ClipboardService;
  var clipboard_component_1 = $__require("9");
  exports.ClipboardComponent = clipboard_component_1.ClipboardComponent;
  
});
})
(function(factory) {
  if (typeof define == 'function' && define.amd)
    define(["@angular/common","@angular/core"], factory);
  else if (typeof module == 'object' && module.exports && typeof require == 'function')
    module.exports = factory(require("@angular/common"), require("@angular/core"));
  else
    throw new Error("Module must be loaded as AMD or CommonJS");
});
