// ng2-clipboard/ng2-clipboard/component/clipboard.component.js
"use strict";
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
Object.defineProperty(exports, "__esModule", { value: true });
var core_1 = require("@angular/core");
var clipboard_service_1 = require("../clipboard.service");
var ClipboardComponent = (function () {
    function ClipboardComponent(clipboard) {
        var _this = this;
        this.clipboard = clipboard;
        this.content = ''; // the text to be copied
        this.noClass = false;
        this.copyText = function () {
            if (_this.content) {
                // console.log('copyText: %s' + this.content);
                _this.clipboard.copy(_this.content);
            }
        };
    }
    // load default config
    ClipboardComponent.prototype.ngOnInit = function () {
        if (!this.faIcon && !this.img && !this.text) {
            this.text = 'copy';
        }
        if (!this.altText) {
            this.altText = 'copy to clipboard';
        }
    };
    return ClipboardComponent;
}());
__decorate([
    core_1.Input(),
    __metadata("design:type", String)
], ClipboardComponent.prototype, "content", void 0);
__decorate([
    core_1.Input(),
    __metadata("design:type", Boolean)
], ClipboardComponent.prototype, "faIcon", void 0);
__decorate([
    core_1.Input(),
    __metadata("design:type", String)
], ClipboardComponent.prototype, "img", void 0);
__decorate([
    core_1.Input(),
    __metadata("design:type", String)
], ClipboardComponent.prototype, "altText", void 0);
__decorate([
    core_1.Input(),
    __metadata("design:type", String)
], ClipboardComponent.prototype, "text", void 0);
__decorate([
    core_1.Input(),
    __metadata("design:type", Boolean)
], ClipboardComponent.prototype, "noClass", void 0);
ClipboardComponent = __decorate([
    core_1.Component({
        // moduleId: module.id,
        selector: 'clipboard-component',
        template: "\n    <span [ngClass]=\"{'clipboard': !noClass}\" (click)=\"copyText()\" [title]=\"altText\">\n      <font-awesome *ngIf=\"faIcon\" [faIcon]=\"'copy'\"></font-awesome>\n      <img *ngIf=\"img\" [src]=\"img\"/>\n    {{text}}</span>\n  ",
        styles: ["\n    .clipboard { cursor: pointer; border-radius: 3px; display: inline-block; padding: 2px; margin: 2px; background-color: #fff; color: #000; box-shadow: 1px 1px 10px black; }\n  "]
    }),
    __metadata("design:paramtypes", [clipboard_service_1.ClipboardService])
], ClipboardComponent);
exports.ClipboardComponent = ClipboardComponent;
